internal/server/users.go

```diff
@@ -137,21 +138,45 @@ func (a *API) CreateUser(c *gin.Context, r *api.CreateUserRequest) (*api.CreateU
 	return resp, nil
 }
 
-func (a *API) UpdateUser(c *gin.Context, r *api.UpdateUserRequest) (*api.User, error) {
-	// right now this endpoint can only update a user's credentials, so get the user identity
-	identity, err := access.GetIdentity(c, data.GetIdentityOptions{ByID: r.ID, LoadProviders: true})
+func (a *API) UpdateUser(c *gin.Context, r *api.UpdateUserRequest) (*api.UpdateUserResponse, error) {
+	rCtx := access.GetRequestContext(c)
+
+	if rCtx.Authenticated.User.ID == r.ID {
+		if err := access.UpdateCredential(c, rCtx.Authenticated.User, r.OldPassword, r.Password); err != nil {
+			return nil, err
+		}
+
+		return &api.UpdateUserResponse{
+			User: *rCtx.Authenticated.User.ToAPI(),
+		}, nil
+	}
+
+	user, err := access.GetIdentity(c, data.GetIdentityOptions{ByID: r.ID, LoadProviders: true})
 	if err != nil {
 		return nil, err
 	}
 
-	err = access.UpdateCredential(c, identity, r.OldPassword, r.Password)
+	password, err := access.ResetCredential(c, user, r.Password)
```

Comment:
I think we might be missing "check password requirements" in this flow as well (admin sets a password for a user).

Reply:
This is intentional. The password reset as implemented by the CLI generates a password which may not conform to the configured password policy. In that case, this request will always fail